### DISCUSSION TITLE
KeyError: 'ActionVisualizer is not in the mmengine::visualizer registry.

### DISCUSSION THREAD

**Discussion by ronniepiku** [open-mmlab/mmaction2#2807] [Q&A]
Hey guys, I'm trying to extract slowonly features from a custom dataset and get this error:

KeyError: 'ActionVisualizer is not in the mmengine::visualizer registry. Please check whether the value of ActionVisualizer is correct or it was registered as expected. More details can be found at https://mmengine.readthedocs.io/en/latest/advanced_tutorials/config.html#import-the-custom-module'

This is the code I am running in an ubuntu terminal:

bash ../../../../mmaction2/tools/dist_test.sh \     slowonly_feature_infer.py     slowonly_imagenet-pretrained-r50_16xb16-8x8x1-steplr-150e_kinetics700-rgb_20221013-15b93b10.pth     $NUM_GPUS --dump result.pkl
I've checked the docs, but I cant find any way to fix this

This is the section of code that returns the error:

log_processor = dict(type='LogProcessor', window_size=20, by_epoch=True)
vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='ActionVisualizer', vis_backends=[dict(type='LocalVisBackend')])
log_level = 'INFO'

**Comment by rendicahya** [comment]
I got the same error message today and it was because my `mmaction/mmaction/visualization` directory was empty. So I put all files from `https://github.com/open-mmlab/mmaction2/tree/main/mmaction/visualization` to that directory and the problem is solved.
